examples/hosts/app-integration/external-data/src/model/taskList.ts

```diff
@@ -292,6 +298,41 @@ export class TaskList extends DataObject implements ITaskList {
 			}
 		}
 	}
+
+	/**
+	 * Register container session data with the customer service.
+	 * @returns A promise that resolves when the registration call returns successfully.
+	 */
+	public async registerWithCustomerService(
+		taskListId: string,
+		containerUrlData: IFluidResolvedUrl | undefined,
+	): Promise<void> {
+		try {
```

Comment:
Maybe not for this change, but one thing to consider is that registering a TaskList with a customer service is a one-way lifecycle event for the TaskList.  Maybe this state should be persisted in the TaskList, and maybe some of its functionality should differ before/after (e.g. should getDraftTasks or writeToExternalServer be allowed to do anything until the registration is done?)

Reply:
Yes I think this is what we want to aim for. I think we shouldn't allow getDraftTasks or writeToExternalServer to do anything until the registration is done.